p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -0,0 +1,76 @@
+package io.p8e.crypto
+
+import com.fortanix.sdkms.v1.ApiClient
+import com.fortanix.sdkms.v1.Configuration
+import com.fortanix.sdkms.v1.api.AuthenticationApi
+import com.fortanix.sdkms.v1.api.SignAndVerifyApi
+import com.fortanix.sdkms.v1.auth.ApiKeyAuth
+import com.fortanix.sdkms.v1.model.DigestAlgorithm
+import com.fortanix.sdkms.v1.model.SignRequest
+import com.fortanix.sdkms.v1.model.VerifyRequest
+import com.google.protobuf.Message
+import io.p8e.proto.Common.Signature
+import io.p8e.util.orThrow
+import io.p8e.util.toProtoUuidProv
+import java.lang.IllegalStateException
+import java.security.KeyPair
+
+class SmartKeySigner(
+    appApiKey: String
+): SignerImpl() {
+    private var sObjUUID: String = ""
+    private val apiKey = appApiKey
+
+    init {
+        val client = ApiClient()
+        client.setBasicAuthString(apiKey)
+        Configuration.setDefaultApiClient(client)
+
+        val authResponse = AuthenticationApi().authorize()
+        val auth = client.getAuthentication("bearerToken") as ApiKeyAuth
+        auth.apiKey = authResponse.accessToken
+        auth.apiKeyPrefix = "Bearer"
+    }
+
+    override fun setKeyId(keyPair: KeyPair) {
+       /*no-op*/
+    }
+
+    override fun setKeyId(uuid: String) {
+        sObjUUID = uuid
+    }
+
+    override fun sign(data: String): Signature = sign(data.toByteArray())
+
+    override fun sign(data: Message): Signature = sign(data.toByteArray())
+
+    override fun sign(data: ByteArray): Signature {
+        val signatureRequest = SignRequest()
+            .hashAlg(DigestAlgorithm.SHA512)
+            .data(data)
+            .deterministicSignature(true)
+
+        val signatureResponse = SignAndVerifyApi().sign(sObjUUID, signatureRequest)
+
+        return Signature.newBuilder()
+            .setAlgo(DigestAlgorithm.SHA512.value)
+            .setProvider("SmartKey")
+            .setSignature(signatureResponse.signature.toString())
+            .setKeyId(signatureResponse.kid.toProtoUuidProv())
+            .build()
+            .takeIf { verify(data, it) }
```

Comment:
do we really need to verify every time we sign?

Reply:
Not sure, I just copied this from the Pen class over, so that each of the signer looks identical, but I don't see any harm in once-overing the signature?

Reply:
I guess it is just one more request against our account. Either way we have to trust that what they send back is accurate, whether it be a signature or a validation result 😆

Reply:
Since we have the public key from smart key, we should be able to sign locally and not have to use the api for it.

Reply:
Yea, we can do that as well, I'll handle it in the next branch.

Reply:
⭐  for @scirner22